src/cmd/singularity/cli/singularity.go

```diff
@@ -38,6 +50,20 @@ func init() {
 	SingularityCmd.Flags().BoolVarP(&quiet, "quiet", "q", false, "Suppress all normal output")
 	SingularityCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Increase verbosity +1")
 
+	usr, err := user.Current()
+	if err != nil {
+		sylog.Fatalf("Couldn't determine user home directory: %v", err)
+	}
+	defaultTokenFile := path.Join(usr.HomeDir, ".singularity", "sylabs-token")
+	// authToken priority default_file < env < file_flag
+	authToken = auth.ReadToken(defaultTokenFile)
```

Comment:
I don't think we should read this before checking for an env / var / flag. The logic in here should stop uneccessary file reads.